FirebaseAdmin/FirebaseAdmin/Auth/Hash/Scrypt.cs

```diff
@@ -0,0 +1,148 @@
+    /// Firebase Auth. See <a cref="StandardScrypt">StandardScrypt</a> for the standard
+    /// Scrypt algorithm. Can be used as an instance of
+    /// <a cref="UserImportHash">UserImportHash</a> when importing users.
+    /// </summary>
+    public sealed class Scrypt : RepeatableHash
+    {
+        private string key;
+
+        private string saltSeparator;
+
+        private int? memoryCost;
+
+        /// <summary>
+        /// Gets or sets the signer key for the hashing algorithm.
+        /// </summary>
+        public string Key
+        {
+            get
+            {
+                if (this.key == null)
+                {
+                    throw new ArgumentNullException("key must be initialized");
+                }
+
+                return this.key;
+            }
+
+            set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentException("key must not be null or empty");
+                }
+
+                var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(value);
+                this.key = UrlSafeBase64Encode(plainTextBytes);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the salt separator for the hashing algorithm.
+        /// </summary>
+        public string SaltSeparator
+        {
+            get
+            {
+                return this.saltSeparator;
+            }
+
+            set
+            {
+                if (value != null)
+                {
+                    var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(value);
+                    this.saltSeparator = UrlSafeBase64Encode(plainTextBytes);
+                }
+                else
+                {
+                    this.saltSeparator = System.Convert.ToBase64String(new byte[0]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the memory cost for the hashing algorithm.
+        /// </summary>
+        public int MemoryCost
+        {
+            get
+            {
+                if (this.memoryCost == null)
+                {
+                    throw new ArgumentNullException("memory cost must be set");
+                }
+
+                return (int)this.memoryCost;
+            }
+
+            set
+            {
+                if (value < 1 || value > 14)
+                {
+                    throw new ArgumentException("memory cost must be between 1 and 14 (inclusive)");
+                }
+
+                this.memoryCost = (int?)value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the hash name which is SCRYPT.
+        /// </summary>
+        protected override string HashName { get { return "SCRYPT"; } }
+
+        /// <summary>
+        /// Gets the minimum number of rounds for a Scrypt hash which is 0.
+        /// </summary>
+        protected override int MinRounds { get { return 1; } }
```

Comment:
Should this be `return 0;`? Or should the comment be set to `1`?

Reply:
Good catch, it should be `return 0`. Thanks!